src/components/TagPopover.js

```diff
@@ -0,0 +1,120 @@
+import React from 'react'
+import { connect } from 'react-redux'
+import A from './A'
+import Avatar from './Avatar'
+import cx from 'classnames'
+import { isEmpty, get } from 'lodash'
+import { hideTagPopover, fetchTagSummary, followTag } from '../actions/index'
+const { string, object, func, array, number, bool } = React.PropTypes
+
+const nounCount = (n, noun) => `${n} ${noun}${Number(n) !== 1 ? 's' : ''}`
+
+@connect(({ tagPopover: { slug, tagName, position, anchorWidth }, tagsByCommunity }) => {
+  let tag = get(tagsByCommunity, [slug, tagName])
+  return {
+    tagName,
+    slug,
+    position,
+    anchorWidth,
+    followed: get(tag, 'followed'),
+    description: get(tag, 'description'),
+    postCount: get(tag, 'post_count'),
+    followerCount: get(tag, 'follower_count'),
+    activeMembers: get(tag, 'active_members')
+  }
+})
+export default class TagPopover extends React.Component {
+
+  static propTypes = {
+    dispatch: func,
+    tagName: string,
+    slug: string,
+    description: string,
+    postCount: string,
+    followerCount: number,
+    activeMembers: array,
+    position: object,
+    anchorWidth: number,
+    followed: bool
+  }
+
+  hide () {
+    let { dispatch } = this.props
+    dispatch(hideTagPopover())
+  }
+
+  removeHideTimeout () {
+    if (this.hideTimeoutId) {
+      clearTimeout(this.hideTimeoutId)
+      this.hideTimeoutId = null
+    }
+  }
+
+  componentWillReceiveProps (nextProps) {
+    let { dispatch, slug, tagName } = this.props
+
+    // hidden
+    if (isEmpty(nextProps.tagName)) return
+
+    // loading
+    if (nextProps.tagName !== tagName || nextProps.slug !== slug) {
+      dispatch(fetchTagSummary(nextProps.tagName, nextProps.slug))
+    }
+
+    // content has loaded
+    if (nextProps.description) {
+      this.hideTimeoutId = setTimeout(() => this.hide(), 3000)
```

Comment:
I used this instead of the document.onMouseover listener, because the latter was not working. It would often be triggered when moving from the tag link to the popover, and so hide the popover before you could get to it.